python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,176 @@
+#
+
+# pylint: disable=wrong-spelling-in-docstring,invalid-name,import-error
+
+""" SparkDLTypeConverters
+Type conversion utilities for definition Spark Deep Learning related MLlib `Params`.
+"""
+
+import six
+
+import tensorflow as tf
+
+from pyspark.ml.param import TypeConverters
+
+import sparkdl.utils.keras_model as kmutil
+
+__all__ = ['SparkDLTypeConverters']
+
+class SparkDLTypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for type conversion functions for :py:func:`Param.typeConverter`.
+    These methods are similar to :py:class:`spark.ml.param.TypeConverters`.
+    They provide support for the `Params` types introduced in Spark Deep Learning Pipelines.
+    """
+
+    @staticmethod
+    def toTFGraph(value):
+        """
+        Convert a value to a :py:obj:`tf.Graph` object, if possible.
+        """
+        if not isinstance(value, tf.Graph):
+            raise TypeError("Could not convert %s to tf.Graph" % type(value))
+        return value
+
+    @staticmethod
+    def asColumnToTensorNameMap(value):
+        """
+        Convert a value to a column name to :py:obj:`tf.Tensor` name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if not isinstance(value, dict):
+            err_msg = "Could not convert [type {}] {} to column name to tf.Tensor name mapping"
+            raise TypeError(err_msg.format(type(value), value))
+
+        # Conversion logic after quick type check
+        strs_pair_seq = []
+        for _maybe_col_name, _maybe_tnsr_name in value.items():
+            # Check if the non-tensor value is of string type
+            _check_is_str(_maybe_col_name)
+            # Check if the tensor name looks like a tensor name
+            _check_is_tensor_name(_maybe_tnsr_name)
+            strs_pair_seq.append((_maybe_col_name, _maybe_tnsr_name))
+
+        return sorted(strs_pair_seq)
+
+    @staticmethod
+    def asTensorNameToColumnMap(value):
+        """
+        Convert a value to a :py:obj:`tf.Tensor` name to column name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if not isinstance(value, dict):
+            err_msg = "Could not convert [type {}] {} to tf.Tensor name to column name mapping"
+            raise TypeError(err_msg.format(type(value), value))
+
+        # Conversion logic after quick type check
+        strs_pair_seq = []
+        for _maybe_tnsr_name, _maybe_col_name in value.items():
+            # Check if the non-tensor value is of string type
+            _check_is_str(_maybe_col_name)
+            # Check if the tensor name looks like a tensor name
+            _check_is_tensor_name(_maybe_tnsr_name)
+            strs_pair_seq.append((_maybe_tnsr_name, _maybe_col_name))
+
+        return sorted(strs_pair_seq)
+
+    @staticmethod
+    def toTFHParams(value):
+        """ Convert a value to a :py:obj:`tf.contrib.training.HParams` object, if possible. """
```

Comment:
let's make this comment correct: "Check that the given value is a :py:obj:`tf.contrib.training.HParams` object and return it. Raises an error if not."